lib/Frontend/FrontendOptions.cpp

```diff
@@ -211,3 +198,254 @@ bool FrontendOptions::isOutputFileDirectory() const {
          llvm::sys::fs::is_directory(getSingleOutputFilename());
 }
 
+const char *FrontendOptions::computeSuffix() {
+  switch (RequestedAction) {
+    case NoneAction:
+      return nullptr;
+      
+    case Parse:
+    case Typecheck:
+    case DumpParse:
+    case DumpInterfaceHash:
+    case DumpAST:
+    case EmitSyntax:
+    case PrintAST:
+    case DumpScopeMaps:
+    case DumpTypeRefinementContexts:
+      // Textual modes.
```

Comment:
These aren't textual; they're just not things that have output that's meaningful to save. (PrintAST and EmitSyntax might be exceptions, but you don't have to change that.)

Reply:
OK, comment is gone!